backends/p4tools/common/core/abstract_execution_state.cpp

```diff
@@ -120,19 +182,54 @@ void AbstractExecutionState::initializeStructLike(const Target &target,
     }
 }
 
-const IR::P4Table *AbstractExecutionState::findTable(const IR::Member *member) const {
-    if (member->member != IR::IApply::applyMethodName) {
-        return nullptr;
-    }
-    if (member->expr->is<IR::PathExpression>()) {
-        const auto *declaration = findDecl(member->expr->to<IR::PathExpression>());
-        return declaration->to<IR::P4Table>();
-    }
-    const auto *type = member->expr->type;
-    if (const auto *tableType = type->to<IR::Type_Table>()) {
-        return tableType->table;
+void AbstractExecutionState::assignStructLike(const IR::StateVariable &left,
+                                              const IR::Expression *right) {
+    if (const auto *structExpr = right->to<IR::StructExpression>()) {
+        std::vector<IR::StateVariable> flatLeftValids;
+        std::vector<const IR::Expression *> flatRightValids;
+        auto flatTargetFields = getFlatFields(left, &flatLeftValids);
+        auto flatStructFields = flattenComplexExpression(structExpr, flatRightValids);
+        auto flatTargetSize = flatTargetFields.size();
+        auto flatStructSize = flatStructFields.size();
```

Comment:
nitpick: I don't see much reason for naming the .size() of a vector, which is calculated in O(1) and is not a very long expression

Reply:
Part of the reason for this naming is just for readability since these variables are used multiple times.